src/NSubstitute/Core/CallRouter.cs

```diff
@@ -90,7 +90,7 @@ private IRoute GetRaiseEventRoute(Func<ICall, object[]> argumentsFactory)
 
         private static bool IsSpecifyingACall(ICall call, IRoute currentRoute)
         {
-            var args = call.GetArguments() ?? EmptyArgs;
+            var args = call.GetOriginalArguments() ?? EmptyArgs;
```

Comment:
Here it doesn't matter. However, decided to use the original arguments for consistency - use them whenever you don't need the modified arguments.